daras_ai_v2/billing.py

```diff
@@ -544,44 +564,90 @@ def render_paypal_subscription_button(
 
 
 def render_payment_information(user: AppUser):
-    assert user.subscription
-
-    gui.write("## Payment Information", id="payment-information", className="d-block")
-    col1, col2, col3 = gui.columns(3, responsive=False)
-    with col1:
-        gui.write("**Pay via**")
-    with col2:
-        provider = PaymentProvider(user.subscription.payment_provider)
-        gui.write(provider.label)
-    with col3:
-        if gui.button(f"{icons.edit} Edit", type="link", key="manage-payment-provider"):
-            raise gui.RedirectException(user.subscription.get_external_management_url())
+    if not user.subscription:
+        return
 
     pm_summary = gui.run_in_thread(
         user.subscription.get_payment_method_summary, cache=True
     )
     if not pm_summary:
         return
```

Comment:
`pm_summary` is often None, and takes a while to load.

When that happens, the cancel button won't be shown.

Reply:
Ok - show this button based on where the user’s subscription has a payment_provider or not?

Reply:
Actually I think this is good. It handles cases where the payment method is saved in stripe/paypal but not in our db?

Reply:
See https://github.com/GooeyAI/gooey-server/blob/9c52e1410c4bb725c49ee272440bbe62192058e5/payments/models.py#L200